core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1176,21 +1198,17 @@ class KafkaApis(val requestChannel: RequestChannel,
   def handleDeleteTopicsRequest(request: RequestChannel.Request) {
     val deleteTopicRequest = request.body.asInstanceOf[DeleteTopicsRequest]
 
-    val (authorizedTopics, unauthorizedTopics) = deleteTopicRequest.topics.asScala.partition( topic =>
-      authorize(request.session, Delete, new Resource(auth.Topic, topic))
+    val (existingAndAuthorizedForDescribeTopics, nonExistingOrUnauthorizedForDescribeTopics) = deleteTopicRequest.topics.asScala.partition( topic =>
+      authorize(request.session, Describe, new Resource(auth.Topic, topic)) && metadataCache.contains(topic)
     )
 
-    val unauthorizedResults = unauthorizedTopics.map ( topic =>
-      // Avoid leaking that the topic exists if the user is not authorized to describe the topic
-      if (authorize(request.session, Describe, new Resource(auth.Topic, topic))) {
-        (topic, Errors.TOPIC_AUTHORIZATION_FAILED)
-      } else {
-        (topic, Errors.INVALID_TOPIC_EXCEPTION)
```

Comment:
I think changing to a UNKNOWN_TOPIC_OR_PARTITION exception might make sense in this case. Especially if that is the behavior we are using for the other requests where a topic is not authorized and we don't want to leak the information. Consistency is always easier to reason about. 

The most important thing is that the exception doesn't give any indication of whether or not the topic exists. Its behavior should the same as if the topic isn't there at all. If this exception is changed, I think the InvalidTopicException in AdminUtils may need to be changed too:
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/AdminUtils.scala#L328